### PR TITLE
[DOC] make some sub-modules DEV only

### PR DESCRIPTION
**Pull request by marehr**
Part of https://github.com/seqan/seqan3/issues/2685

### CODE REVIEW

File: include/seqan3/utility/parallel/all.hpp [hunk 1]
```diff
@@ -30,6 +32,7 @@
  *
  * \endif
```
Comment (SGSSGene) [inline include/seqan3/utility/parallel/all.hpp:33]: (Can't mark the correct lines)
Should the `\if DEV` block from line 28-33 be removed?

**Review by SGSSGene — CHANGES_REQUESTED** [review]

File: include/seqan3/utility/parallel/all.hpp (hunk 1 above)
Comment (eseiler) [reply include/seqan3/utility/parallel/all.hpp:33]: I'd even say that just the whole file should just be `\\!cond DEV`ed?

Comment (marehr) [reply include/seqan3/utility/parallel/all.hpp]: like so?

Comment (eseiler) [reply include/seqan3/utility/parallel/all.hpp]: if you put the cond above `/*!\file`, it shouldn't show up at all? 
https://seqan3-git-fork-marehr-devonly-seqan.vercel.app/usr/html/dir_8af16cb5769d23173486b3b2f0ac1c5d.html
it shows all the files.

**Review by SGSSGene — APPROVED** [review]
LGTM!

File: include/seqan3/utility/parallel/all.hpp [hunk 2]
```diff
@@ -6,12 +6,14 @@
 // -----------------------------------------------------------------------------------------------------
```
Comment (eseiler) [inline include/seqan3/utility/parallel/all.hpp:7]: Can you try putting `\\!cond DEV` here?

**Review by eseiler — CHANGES_REQUESTED** [review]
Can you try excluding everything?

**Comment by eseiler** [comment]
> Can you try excluding everything?

Actually, that doesn't help.
What works would be

```patch
diff --git a/test/documentation/seqan3_doxygen_cfg.in b/test/documentation/seqan3_doxygen_cfg.in
index 9829e2f44..47653493b 100644
--- a/test/documentation/seqan3_doxygen_cfg.in
+++ b/test/documentation/seqan3_doxygen_cfg.in
@@ -68,7 +68,9 @@ HTML_DYNAMIC_SECTIONS  = YES
 ## detect headers without extensions (in std module)
 EXTENSION_MAPPING      = .no_extension=C++
 FILE_PATTERNS          = *
-EXCLUDE_PATTERNS       = */doc/*.cpp
+EXCLUDE_PATTERNS       = */doc/*.cpp \
+                         */utility/parallel/* \
+                         */alignment/matrix/*
 
 ## Developer VS user mode
 EXTRACT_PRIVATE        = ${SEQAN3_DOXYGEN_EXTRACT_PRIVATE}

```

Then you don't need the `\\!cond`s, and it is completely hidden.

We could even do `*/detail/*` for usr documentation, but then we would have to figure out what to do with `include/seqan3/core/detail/customisation_point.hpp` as the macros there are currently expanded by doxygen.

**Review by eseiler — APPROVED** [review]

**Pull request merged**